wfe2/verify.go

```diff
@@ -292,11 +294,17 @@ func (wfe *WebFrontEndImpl) matchJWSURLs(outer, inner *jose.JSONWebSignature) *p
 	return nil
 }
 
+// bJSONWebSignature wraps a *jose.JSONWebSignature, but ensures it has been
+// created and processed by the various safety/sanity checks in wfe.parseJWS.
+type bJSONWebSignature struct {
+	*jose.JSONWebSignature
+}
```

Comment:
I'm pretty sure this could also just be `type bJSONWebSignature jose.JSONWebSignature`. They both have the same effect -- declaring a new type name and describing what that values of that type should look like -- but yours adds an extra struct wrapper that isn't necessary in this case.

Reply:
Going the type definition route you suggested ends up in an error here:
```
	parsedJWS, err := jose.ParseSigned(bodyStr)
        ...
	return &bJSONWebSignature{parsedJWS}, nil
```

The error being: 
> cannot use parsedJWS (variable of type *jose.JSONWebSignature) as []jose.Signature value in struct literal compiler IncompatibleAssign


If I stay on the embed path then this works as intended.

Reply:
You want `return &bJSONWebSignature(parsedJWS), nil`, I think. You can't use the curly braces there because that's saying "create a bJSONWebSignature struct, then assign parsedJWS to be the value of its only struct member.

Reply:
Here's a very important distinction of when someone would want to either use a type definition vs struct embedding. A new type created by type definition does not inherit methods on the concrete type. A struct that embeds a type does inherit those methods. In a few places we need to access `jws.Verify(jwk)` which currently is inaccessible.
```
jws.Verify undefined (type *bJSONWebSignature has no field or method Verify)compiler[MissingFieldOrMethod](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#MissingFieldOrMethod)
```

Reply:
Ooooh that is a very good distinction!